cmd/zed/lake/index/add.go

```diff
@@ -0,0 +1,119 @@
+package index
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+
+	zedlake "github.com/brimdata/zed/cmd/zed/lake"
+	"github.com/brimdata/zed/pkg/charm"
+	"github.com/brimdata/zed/pkg/rlimit"
+	"github.com/segmentio/ksuid"
+)
+
+var Add = &charm.Spec{
+	Name:  "add",
+	Usage: "add [options] [ -rule ruleid ] tag [ ...tag ]",
+	Short: "index one or more tags",
+	New:   NewAdd,
+}
+
+type AddCommand struct {
+	lake   *zedlake.Command
+	commit bool
+	rids   idArray
+	zedlake.CommitFlags
+}
+
+func NewAdd(parent charm.Command, f *flag.FlagSet) (charm.Command, error) {
+	c := &AddCommand{lake: parent.(*Command).Command}
+	f.BoolVar(&c.commit, "commit", false, "commit added index rule if successfully written")
+	f.Var(&c.rids, "rule", "id of rule(s) to index (can be specified multiple times")
```

Comment:
Maybe simplify by dropping `idArray`, changing the type of `AddCommand.rid` to `[]ksuid.KSUID` and using `flag.FlagSet.Func`?
```suggestion
	f.Func("rule", "id of rule to index (can be specified multiple times", func(s string error {
		id, err := zedlake.ParseID(s)
		c.rids = append(c.rids, id)
		return err
	})
```